datastream-kafka/src/test/java/com/linkedin/datastream/kafka/TestKafkaProducerWrapper.java

```diff
@@ -150,7 +151,7 @@ void verifySend(int numExpected) {
     }
 
     void verifyFlush(int numExpected) {
-      verify(_mockProducer, times(numExpected)).flush();
+      verify(_mockProducer, times(numExpected)).flush(anyInt(), any(TimeUnit.class));
```

Comment:
Do you see a value in adding a test to ensure that flush timeout is honored?

Reply:
One example is here https://github.com/linkedin/brooklin/pull/676/files where it can throw timeoutException. See if this is useful.

Reply:
As discussed, adding a test similar to what you had in that PR is hard because we are relying on Kafka's internal timeout  mechanism, unlike in your earlier PR where the `KafkaProducerWrapper` itself was responsible for timing out the  flush. Instead, added a test to  validate that we handle TimeoutException as expected.